File: TaskBarProgressBarC#/Class1.cs
using Microsoft.VisualBasic;
using Microsoft.WindowsAPICodePack.Taskbar;
using System;
using System.Windows.Forms;

namespace TaskBarProgressBarC_
{
    public partial class TaskBarProgressBar
    {
        private static Form condenatedform;
        private static readonly TaskbarManager windowsTaskbar = TaskbarManager.Instance;
        public enum State
        {
            None,
            Normal,
            Pause,
            Error,
            Undefined
        }
        public static object SetState(State taskstate, Form form)
        {
            if (taskstate == State.None)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.NoProgress, form.Handle);
            }
            else if (taskstate == State.Normal)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Normal, form.Handle);
            }
            else if (taskstate == State.Pause)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Paused, form.Handle);
            }
            else if (taskstate == State.Error)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Error, form.Handle);
            }
            else if (taskstate == State.Undefined)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Indeterminate, form.Handle);
            }
            condenatedform = form;
            return condenatedform;
        }
        public static bool SetProgress(int value, int maxvalue)
        {
            try
            {
                windowsTaskbar.SetProgressValue(value, maxvalue, condenatedform.Handle);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message, MsgBoxStyle.Critical, "No se inicio el estado!");
            }
            return true;
        }
        public static bool SetProgress(int value, int maxvalue, Form form, State taskstate)
        {
            if (taskstate == State.None)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.NoProgress, form.Handle);
            }
            else if (taskstate == State.Normal)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Normal, form.Handle);
            }
            else if (taskstate == State.Pause)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Paused, form.Handle);
            }
            else if (taskstate == State.Error)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Error, form.Handle);
            }
            else if (taskstate == State.Undefined)
            {
                windowsTaskbar.SetProgressState(TaskbarProgressBarState.Indeterminate, form.Handle);
            }
            windowsTaskbar.SetProgressValue(value, maxvalue, form.Handle);
            return true;
        }
    }
}

File: TaskBarProgressBar/Class1.vb
Imports System.Windows.Forms
Imports Microsoft.WindowsAPICodePack.Taskbar

Public Class TaskBarProgressBar
    Private Shared condenatedform As Form
    Private Shared ReadOnly windowsTaskbar As TaskbarManager = TaskbarManager.Instance
    Enum State
        None
        Normal
        Pause
        [Error]
        Undefined
    End Enum
    Public Shared Function SetState(taskstate As State, form As Form)
        If taskstate = State.None Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.NoProgress, form.Handle)
        ElseIf taskstate = State.Normal Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Normal, form.Handle)
        ElseIf taskstate = State.Pause Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Paused, form.Handle)
        ElseIf taskstate = State.Error Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Error, form.Handle)
        ElseIf taskstate = State.Undefined Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Indeterminate, form.Handle)
        End If
        condenatedform = form
        Return condenatedform
    End Function
    Public Shared Function SetProgress(value As Integer, maxvalue As Integer) As Boolean
        Try
            windowsTaskbar.SetProgressValue(value, maxvalue, condenatedform.Handle)
        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Critical, "No se inicio el estado!")
        End Try
        Return True
    End Function
    Public Shared Function SetProgress(value As Integer, maxvalue As Integer, form As Form, taskstate As State) As Boolean
        If taskstate = State.None Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.NoProgress, form.Handle)
        ElseIf taskstate = State.Normal Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Normal, form.Handle)
        ElseIf taskstate = State.Pause Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Paused, form.Handle)
        ElseIf taskstate = State.Error Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Error, form.Handle)
        ElseIf taskstate = State.Undefined Then
            windowsTaskbar.SetProgressState(TaskbarProgressBarState.Indeterminate, form.Handle)
        End If
        windowsTaskbar.SetProgressValue(value, maxvalue, form.Handle)
        Return True
    End Function
End Class